python/ray/data/dataset_pipeline.py

```diff
@@ -83,6 +83,7 @@ def __init__(
         stages: List[Callable[[Dataset[Any]], Dataset[Any]]] = None,
         length: int = None,
         progress_bars: bool = progress_bar._enabled,
+        base_datasets_can_be_cleared: bool = False,
```

Comment:
Can we actually make this a required variable? Otherwise it's easy to forget to set this properly.